spec/models/tag_association_spec.rb

```diff
@@ -0,0 +1,52 @@
+# == Schema Information
+#
+# Table name: tag_associations
+#
+#  id          :integer          not null, primary key
+#  from_tag_id :integer          not null
+#  to_tag_id   :integer          not null
+#  created_at  :datetime
+#  updated_at  :datetime
+#
+
+require 'spec_helper'
+
+describe TagAssociation do
+  let!(:mainstream_browse_page) { create(:mainstream_browse_page) }
+  let!(:topic)                  { create(:topic) }
+
+  it "is created by associating mainstream browse pages and topics" do
+    expect(TagAssociation.where(from_tag: mainstream_browse_page,
+                                to_tag: topic).size).to eq(0)
+
+    mainstream_browse_page.topics << topic
+    expect(mainstream_browse_page.save).to be_true
+
+    expect(TagAssociation.where(from_tag: mainstream_browse_page,
+                                to_tag: topic).size).to eq(1)
+  end
+
+  describe "cascade destroying associated tags" do
+    it "destroys associated topics when topic destroyed" do
+      mainstream_browse_page.topics << topic
+      expect(mainstream_browse_page.save).to be_true
+
+      expect(mainstream_browse_page.topics).to_not be_empty
+      expect(topic.destroy).to be_true
+
+      mainstream_browse_page.reload
+      expect(mainstream_browse_page.topics).to be_empty
+    end
+
+    it "destroys associated topics when mainstream browse page destroyed" do
```

Comment:
The wording of this is a little unclear, it suggests that the topics themselves should be destroyed...  Maybe "destroys the topic associations when a mainstream browse page is destroyed"

Similarly for the spec above...